explainer-rendering.md

```diff
@@ -0,0 +1,110 @@
+Formatted Text - Rendering
+=============
+An approach to rendering multi-line formatted text to an HTML canvas context.
+
+This explainer focuses on rendering [the data model](explainer-datamodel.md) for formatted text.
+For a general overview of the problem space, see the repo's [readme](readme.md). In addition to rendering,
+the data model can [provide metrics](explainer-metrics.md).
+
+---
+## Specifying the layout constraints
+
+## Initial values for CSS properties, esp. when used with the Canvas. Who wins?
+
+## CSS unit value resolution (percentages, px, vh, etc.)
+
+
+
+Next, to render the text into multiple lines, we use the new `fillFormattedText` API
+which takes a max width to wrap the text at.
+
+```js
+context.font = "18pt Arial";
+context.fillFormattedText( formattedText, /*x*/0, /*y*/50, /*wrapWidth*/250 );
+```
+
+This would produce the following output on the canvas:
+
+<img src="explainerresources/Example1.png" alt="Wrapped text rendered in a canvas." align="center"/>
+
+### Bidi Text
+
+No additional work is needed from web developers to support bidi text.
+Implementations will perform bidi analysis on the `CanvasFormattedText`'s text runs
+and create internal bidi runs if necessary. An example demonstrating bidi text follows:
+
+```js
+const context = document.getElementById( "myCanvas" ).getContext( "2d" );
+context.font = "30px Arial";
+let canvasFormattedText = new CanvasFormattedText();
+canvasFormattedText.appendRun( { text: "Sample arabic بمدينة مَايِنْتْس، ألمانيا text." } );
+context.fillFormattedText( canvasFormattedText, /*x*/0, /*y*/30, /*wrapWidth*/350 );
+```
+
+produces the following output on the canvas:
+
+<img src="explainerresources/Example2.png" alt="Wrapped text rendered in a canvas." align="center"/>
+
+### Pre-existing text controls
```

Comment:
Only the default font is carried over from the canvas context - since nothing will render without a font defined. Can we simplify this section to say only the font is inherited.

Reply:
Given this, I don't suppose we should rename the API to `drawFormattedText` since we're not technically _stroking_ it or _filling_ it? Exercise for a future spec update.

Reply:
I have other questions here that need testing to resolve. For example, I suspect that the default transform is also applied at render time and probably more. Might need to be more specific on what we mean here.

Reply:
See updates https://github.com/WICG/canvas-formatted-text/commit/0686b82c39e25fd9692e17f47c278a6afa3b680d